Remove StdModels related code

lukstbit opened the pull request:
<!--- Please fill the necessary details below -->
## Purpose / Description

Fixed a TODO about removing StdModels. See commit message for more info.

## How Has This Been Tested?

Ran the app, Ran the tests.

## Checklist

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

Arthur-Milchior commented on AnkiDroid/src/test/java/com/ichi2/libanki/CollectionTest.kt:
```diff
@@ -172,10 +174,13 @@ class CollectionTest : JvmTest() {
+        // old code used StdModels.STD_MODELS.size for this variable. There were 6 models:
+        // BASIC_MODEL, BASIC_TYPING_MODEL, FORWARD_REVERSE_MODEL, FORWARD_OPTIONAL_REVERSE_MODEL,
+        // CLOZE_MODEL, IMAGE_OCCLUSION_MODEL
+        val stdModelSize = 6
```
Not fan of this variable name. I'd go with `numberOfStandardModel`. 
I know this commit don't create this variable, but since we're touching it anyway, maybe we can improve, and note that it's note the size of a model.

Arthur-Milchior replied:
I'd suggest copying code from `AddNewNotesType`, using 
```
        val stdModelSize =
            StockNotetype.Kind.entries.count { it != StockNotetype.Kind.UNRECOGNIZED }
```

Arthur-Milchior commented on AnkiDroid/src/test/java/com/ichi2/libanki/CollectionTest.kt:
```diff
@@ -172,10 +174,13 @@ class CollectionTest : JvmTest() {
         assertEquals(col.notetypes.all().size, stdModelSize)
         for (i in 0..99) {
-            StdModels.BASIC_MODEL.add(col)
+            col.createBasicModel(BASIC_MODEL_NAME)
```
Since we always use the name `BASIC_MODEL_NAME` maybe we can remove the parameter and hard code it in the function. I doubt we'll ever use other name anyway

lukstbit replied:
I added it as a default parameter value in createBasicModel() so it makes the call site cleaner.

Arthur-Milchior commented on AnkiDroid/src/test/java/com/ichi2/utils/NoteTypeTest.kt:
```diff
@@ -84,3 +89,44 @@ class NoteTypeTest {
+ * @return the new model
+ */
+fun Collection.createBasicModel(name: String): NotetypeJson {
+    val m = notetypes.new(name)
```
Maybe we could reuse code from AddNoteType and define it as
```            
col.addNotetypeLegacy(col.getStockNotetypeLegacy(StockNotetype.Kind.KIND_BASIC))
```

(This assumes that we keep the name constant), which is currently the case. If we want to keep the parameter `name` the change is easy to do)

lukstbit replied:
Done. Probably not needed, but I kept the name just to be more flexible in the future if the method gets more usage.

Arthur-Milchior commented on AnkiDroid/src/androidTest/java/com/ichi2/anki/tests/ContentProviderTest.kt:
```diff
@@ -111,6 +113,23 @@ class ContentProviderTest : InstrumentedTest() {
     }
 
+    private fun createBasicModel(name: String = BASIC_MODEL_NAME): NotetypeJson {
+        val m = col.notetypes.new(name)
```
May I suggest

```
    private fun createBasicModel(name: String = BASIC_MODEL_NAME): NotetypeJson {
        val m =
            BackendUtils.from_json_bytes(col.getStockNotetypeLegacy(
                StockNotetype.Kind.KIND_BASIC)).apply {
                    set("name", name)
            }
        col.addNotetypeLegacy(BackendUtils.to_json_bytes(m))
        return col.notetypes.byName(name)!!
    }
```

This would be closer to current implementation, using anki's backend where we can.

lukstbit replied:
Much cleaner, done!

Arthur-Milchior reviewed (commented):
I'm fine with the change as-is, but think it may be slightly improved, deleting even more code.

lukstbit commented on AnkiDroid/src/test/java/com/ichi2/utils/NoteTypeTest.kt:
```diff
@@ -84,3 +89,44 @@ class NoteTypeTest {
+ */
+fun Collection.createBasicModel(name: String): NotetypeJson {
+    val m = notetypes.new(name)
+    val frontName = AnkiDroidApp.appResources.getString(R.string.front_field_name)
```
They are also used in CardTemplateEditor in a user facing position so they can't be deleted yet.

BrayanDSO reviewed (approved):
Nice!

lukstbit commented:
I fixed the conflict.
I addressed Arthur's comments and with the two approvals I'm merging this myself along with #16746 so I don't have to do several string syncs as well.
